examples/plot_missing_values.py

```diff
@@ -1,73 +1,117 @@
+    missing_features = rng.randint(0, n_features, n_missing_samples)
+
+    # Estimate the score without the lines containing missing values
+    X_filtered = X_full[~missing_samples, :]
+    y_filtered = y_full[~missing_samples]
+    estimator = RandomForestRegressor(random_state=0, n_estimators=100)
+    subset_score = cross_val_score(estimator, X_filtered, y_filtered,
+                                   scoring='neg_mean_squared_error').mean()
+
+    # Estimate the score after imputation (mean strategy) of the missing values
+    X_missing = X_full.copy()
+    X_missing[np.where(missing_samples)[0], missing_features] = 0
+    y_missing = y_full.copy()
+    estimator = Pipeline([("imputer", Imputer(missing_values=0,
+                                              strategy="mean",
+                                              axis=0)),
+                          ("forest", RandomForestRegressor(random_state=0,
+                                                           n_estimators=100))])
+    mean_impute_score = cross_val_score(estimator, X_missing, y_missing,
+                                        scoring='neg_mean_squared_error'
+                                        ).mean()
+
+    # Estimate the score after imputation (MICE strategy) of the missing values
+    estimator = Pipeline([("imputer", MICEImputer(missing_values=0,
+                                                  random_state=0)),
+                          ("forest", RandomForestRegressor(random_state=0,
+                                                           n_estimators=100))])
+    mice_impute_score = cross_val_score(estimator, X_missing, y_missing,
+                                        scoring='neg_mean_squared_error'
+                                        ).mean()
+
+    return full_score, subset_score, mean_impute_score, mice_impute_score
+
+
+mses_diabetes = np.array(get_results(load_diabetes())) * -1
+mses_boston = np.array(get_results(load_boston())) * -1
+
+plt.figure(figsize=(12, 6))
+xval = np.arange(len(mses_diabetes))
+labels = {0: ('MSE with entire dataset', 'r'),
+          1: ('MSE excluding samples containing missing values', 'g'),
+          2: ('MSE after mean imputation of the missing values', 'b'),
+          3: ('MSE after MICE imputation of the missing values', 'orange')}
+
+# plot diabetes results
+ax1 = plt.subplot(121)
+for j in range(len(xval)):
+    label, color = labels[xval[j]]
+    ax1.bar(xval[j], mses_diabetes[j], width=0.8, color=color, alpha=0.6,
+            align='center', label=label)
+ax1.legend(loc='upper left')
+ax1.set_ylabel('Mean Squared Error')
+ax1.set_title('Feature Selection Techniques with Diabetes Data')
+ax1.set_ylim(bottom=np.min(mses_diabetes) * 0.9,
+             top=np.max(mses_diabetes) * 1.15)
+
+# plot boston results
+ax2 = plt.subplot(122)
+for j in range(len(xval)):
+    label, color = labels[xval[j]]
+    ax2.bar(xval[j], mses_boston[j], width=0.8, color=color, alpha=0.6,
```

Comment:
Could you also show error bars? The MSE reduction when removing samples containing missing values is disconcerting. (Although I realise it's also because the test data doesn't have missing values which should be noted somewhere. Or perhaps we should instead report "Missing values replaced by 0"...?)

I think this chart should be transposed so that you can give the label for each bar as the yticklabel.

Reply:
I like "Missing values replaced by 0" - much clearer story. Will do all the other stuff too.